src/bootstrap/install.rs

```diff
@@ -226,14 +225,39 @@ install!((self, builder, _config),
         });
         install_analysis(builder, self.stage, self.target);
     };
-    Src, "src", Self::should_build(_config) , only_hosts: true, {
-        builder.ensure(dist::Src);
-        install_src(builder, self.stage);
-    }, ONLY_BUILD;
     Rustc, "src/librustc", true, only_hosts: true, {
         builder.ensure(dist::Rustc {
             compiler: builder.compiler(self.stage, self.target),
         });
         install_rustc(builder, self.stage, self.target);
     };
 );
+
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct Src {
+    pub stage: u32,
+}
+
+impl Step for Src {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
```

Comment:
Hm doesn't this mean that we'll be producing the dist-src package on all dist builders now?

Reply:
This is the install step; we don't run install on CI. The `dist::Src` step hasn't changed the conditions it runs on. Neither has this step; I just pulled it out because the macro creates the struct with too many fields for Src... I can go into more detail if necessary, it's somewhat complicated.

Reply:
Er right sorry I mixed this up with `dist::Src`, my bad!